dashboard/app/models/user.rb

```diff
@@ -619,9 +619,14 @@ def user_level_locked?(script_level, level)
   # script that hasn't yet been passed, starting its search at the last level we submitted
   def next_unpassed_progression_level(script)
     user_levels_by_level = user_levels_by_level(script)
+    user_levels = user_levels_by_level.values.flatten.select do |user_level|
```

Comment:
One of the cases I was looking at was an instance where user_level.script_id was actually for a different script than any of the user_levels. I must admit, I dont understand this scenario:
```
irb(main):062:0> UserLevel.find(985612048).script_id
=> 142
irb(main):063:0> UserLevel.find(985612048).level.script_levels.map{|sl| sl.script_id}
=> [230]
```